Automattic/wp-calypso
# Framework: Remove abtest library

#### Changes proposed in this Pull Request

This PR removes the last usages of its abtest library, along with the library itself. 

For context and rationale, see pbmo2S-LH-p2 and the comment thread there.

This PR is remotely related to #24004 where we aim to reduxify `lib/user` - it makes the migration easier by avoiding unnecessary work to migrate the abtesting library.

#### Testing instructions

* Verify Calypso builds and all tests still pass.
* Do a thorough smoke test of Calypso and verify everything works the same as before. Make sure to try:
  * Logging in and out
  * Signing up and creating a new site
  * Purchasing a plan and upgrading to a higher-tier plan
